src/neo4j_graphrag/generation/graphrag.py

```diff
@@ -83,6 +87,7 @@ def __init__(
     def search(
         self,
         query_text: str = "",
+        chat_history: Optional[list[dict[str, str]]] = None,
```

Comment:
`chat_history` is an optional list of type `{"role": "system"|"assistant"|"user", "content": str}`, which usually are referred to as `messages`, so I wonder if we can 

1. be more specific in the type declaration 
2. and also rename the argument (everywhere) to `message_history`

Might seem as a unnecessary nitpick, but the messages might not come from a chat.

Reply:
Rename: [37225fd](https://github.com/neo4j/neo4j-graphrag-python/pull/234/commits/37225fdd124d21701cbfa7a0f7256c3d91e8c3b4), type declaration: [66b2358](https://github.com/neo4j/neo4j-graphrag-python/pull/234/commits/66b235872500fadbe727d6983695571146419e83).

Reply:
Tried to declare the type as [BaseMessage](https://github.com/leila-messallem/neo4j-graphrag-python/blob/a749a9e2eb596763d79c6375170be9bc8b0478d0/src/neo4j_graphrag/llm/types.py#L9), but this doesn't work out if we want to receive a dictionary. So how exactly do you mean that we can achieve a more specific type declaration?

Reply:
This should do it, if I understand your question.

```python
from typing import TypedDict, Literal

class Message(TypedDict):
    role: Literal["system", "user", "assistant"]
    content: str
```

That would make this pass the type checker:

```python
message: Message = {"role": "user", "content": "Hello, world!"}
```

but not 

```python
invalid_message: Message = {"role": "bot", "content": "Invalid role"}
```

Reply:
How is this different from abef33c](https://github.com/neo4j/neo4j-graphrag-python/pull/234/commits/abef33c195b7c23ec0a18482baa4190775fe6b10)?

Reply:
The pydantic model subclass needs to be instantiated like `message = BaseMessage(role="user", content="Hello!")` where the typings type can be a plain dict `message: Message = {"role": "user", "content": "Hello, world!"}`.

The difference is that the latter is easier to integrate with when relaying data that already are plain dicts, but there's no runtime validation.

Reply:
So you want to replace the pydantic model with a TypedDict and scrap [the validation](https://github.com/leila-messallem/neo4j-graphrag-python/blob/17db6b18de6708ffac48fb557d5991b387d2fac2/src/neo4j_graphrag/llm/openai_llm.py#L79)?

Reply:
On the specific line I commented on, the typing is not helpful for developers `Optional[list[dict[str, str]]]`.

Reply:
Ok, I get it now, [d5a287b](https://github.com/neo4j/neo4j-graphrag-python/pull/234/commits/d5a287b94aaeb33bbfe9634e8b2715fd26fdab38). 

Mypy doesn't accept it, but I guess it's something you work around with fake "casting"?